inform7/Internal/Inter/Architecture32Kit/Sections/Startup.i6t

```diff
@@ -2,176 +2,121 @@ Startup Template.
 
 How programs for this architecture start up the virtual machine.
 
-@ This variable is set at startup, to avoid asking the interpreter over
-and over again.
+@ |VM_Check_Functionality| checks that we are running in a Glk interpreter,
+as well as that it supports the minimum requirements, which are currently
+that Glk supports unicode, and Glulx supports real numbers. If one of those
+requirements is not met an error message will be shown before quitting.
 
 =
-Global unicode_gestalt_ok; ! Set if interpreter supports Unicode
+[ VM_Check_Functionality res;
+	@gestalt 4 2 res; ! Test if this interpreter has Glk...
+	if (res == 0) quit; ! ...without which there would be nothing we could do
 
-@ |VM_Initialise()| is almost the first routine called, except that the
-"starting the virtual machine" activity is allowed to go first; and,
-come to think of it, memory allocation has to be set up before even that,
-and that in turn calls |VM_PreInitialise()| to do the absolute minimum.
+	! Set the VM's I/O system to be Glk.
+	@setiosys 2 0;
 
-Arrangements are a little different here compared with the 16-bit architecture,
-because some data is retained in the case of a restart.
-
-(Many thanks are due to Eliuk Blau, who found several tricky timing errors
-here and elsewhere in the Glulx-specific code. Frankly, I feel like hanging
-a sign on the following routines which reads "Congratulations on bringing
-light to the Dark Room.")
-
-=
-[ VM_PreInitialise res;
-    @gestalt 4 2 res; ! Test if this interpreter has Glk...
-    if (res == 0) quit; ! ...without which there would be nothing we could do
+	if (~~glk_gestalt(gestalt_Unicode, 0)) {
+		Fatal_Error("This storyfile can only be run in an unicode interpreter.");
```

Comment:
This is a functional change: rather than raising a RTP in `TEXT_TY_CastPrimitive` it will error immediately. But this seems reasonable as there are many other calls of _uni functions that don't check if unicode is supported. `unicode_gestalt_ok` was only ever actually checked in `TEXT_TY_CastPrimitive`.